File: csc330/poundsPractice/Drink.py
import math

class Drink:

    MAXIMUM_CAPACITY = 10
    DEFAULT_COST = 2.00

    def __init__(self, brandName, cost = None):
      self.name = brandName
      self.consumed = 0
      self.available = 0
      self.moneyMade = 0
      if cost is None:
        self.drinkCost = Drink.DEFAULT_COST
        self.useDefaultCost = 1
        self.useSuppliedCost = 0
      else:
        self.drinkCost = cost
        self.useDefaultCost = 0
        self.useSuppliedCost = 1

    def refill(self):
        self.available = Drink.MAXIMUM_CAPACITY

    def vend(self):
        if (self.available > 0):
            self.available = self.available - 1
            self.consumed = self.consumed + 1
            self.moneyMade = self.moneyMade + (
                    self.useDefaultCost * Drink.DEFAULT_COST +
                    self.useSuppliedCost * self.drinkCost )
            print( self.name + " purchased.")
        else:
            print("************")
            print("* SOLD OUT *")
            print("************")


    def profit(self):
        return round(self.moneyMade * 100.0)/100.0

    def drinksSold(self):
        return self.consumed

    def getDrinkName(self):
        return self.name

    def restockAmount(self):
        return Drink.MAXIMUM_CAPACITY - self.available

File: csc330/poundsPractice/Drink.cs
using System;

public class Drink {

// the following will be class variables

   private const int MAXIMUM_CAPACITY = 10;
   private const double DEFAULT_DRINK_COST = 1.75;

   private int available, consumed;
   private int useDefaultCost, useSuppliedCost;
   private double drinkCost, moneyMade;
   private String name;
  
// Start defining constructors here...

// This constructor sets the name of the drink only.
 
   public Drink (String brandName) {
      name = brandName;
      consumed = 0;
      available = 0;
      moneyMade = 0.0;
      drinkCost = DEFAULT_DRINK_COST;
      useDefaultCost = 1;
      useSuppliedCost = 0; 
   }

   public Drink (String brandName, double cost) {
      name = brandName;
      consumed = 0;
      available = 0;
      moneyMade = 0.0;
      drinkCost = cost;
      useDefaultCost = 0;
      useSuppliedCost = 1; 
   }


// Now start defining class methods

// This is the basic vend method.  
   public void vend () {
      if ( available > 0 ) {
         available -= 1;
         consumed += 1;
         moneyMade += ( useDefaultCost * DEFAULT_DRINK_COST +
                        useSuppliedCost * drinkCost ); 
         Console.WriteLine(name + " Puchased.");
         }
      else {
         Console.WriteLine("************");
         Console.WriteLine("* SOLD OUT *");
         Console.WriteLine("************");
        } 
     } 


// Add a "setter" function called fillDrinks to set the number of
// drinks to their maximum capacity.
   public void refill () {
      available = MAXIMUM_CAPACITY;
      }


   public double profit () {
      return Math.Round(moneyMade*100.0)/100.0;
      }


   public int drinksSold () {
      return consumed;
      }


   public String getDrinkName () {
      return name;
      }

   public int restockAmount () {
       return MAXIMUM_CAPACITY - available;
   }
}